app/vmalert/rule/alerting.go

```diff
@@ -274,24 +275,32 @@ func (ar *AlertingRule) toLabels(m datasource.Metric, qFn templates.QueryFn) (*l
 		return nil, fmt.Errorf("failed to expand labels: %w", err)
 	}
 	for k, v := range extraLabels {
-		ls.processed[k] = v
+		// if conflicted, reserve origin labels in ls.origin,
+		// add prefix `exported_` for origin labels in ls.processed.
 		if _, ok := ls.origin[k]; !ok {
 			ls.origin[k] = v
+		} else {
+			ls.processed[fmt.Sprintf("exported_%s", k)] = ls.processed[k]
 		}
+		ls.processed[k] = v
 	}
 
 	// set additional labels to identify group and rule name
 	if ar.Name != "" {
-		ls.processed[alertNameLabel] = ar.Name
 		if _, ok := ls.origin[alertNameLabel]; !ok {
 			ls.origin[alertNameLabel] = ar.Name
+		} else {
```

Comment:
I think we should override them anyway and maybe allow user to disable this by adding [something](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5398#issuecomment-1827740459) like `-disableExportedLabel`, similar to `honorLabels` in scrape_config.

Reply:
Now it maybe weird when vmalert has `externalLabel` and there are rules chained, then `externalLabel` will be added twice as `externalLabel: ` and `exported_externalLabel: `, see this [test](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/2403cf217da0796626b9f63fe2bc29313ae9d68a/app/vmalert-tool/unittest/testdata/test1.yaml#L69) example.

Reply:
What if we check if values of conflicting labels are actually different before using `exported_` logic?
```
if ls.processed[k] != v {
  ls.processed[fmt.Sprintf("exported_%s", k)] = ls.processed[k]
}
```

This would prevent cases like this https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5398#discussion_r1417559460 and this https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5398#discussion_r1430190711

Reply:
Sure, please check https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5398/commits/7e8f3b039e9fca8d1cd9811438c97990f78d3d7f